### ISSUE TITLE
TreeGrid: set initial tabindex attributes for the roving tabindex

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#69320]
### Description

Noticed while working on https://github.com/WordPress/gutenberg/issues/69147 / https://github.com/WordPress/gutenberg/pull/69190

The TreeGrid roving tabindex inplementation is within the [RovingTabIndexItem](https://github.com/WordPress/gutenberg/blob/7a4b5aa5d8a63d97b084e1339e6ce544dfc944bb/packages/components/src/tree-grid/roving-tab-index-item.tsx) component and works in a way that the `tabinxed` attributes are set only when one of the items in the tree grid receives focus.

This means that on initial first rendering, _all the items are focusable_. I'm not sure this is ideal because it may lead to problems when _other components_ manage focus on the tree grid or other components that wrap it, as noticed in https://github.com/WordPress/gutenberg/issues/69147.

To remind what an ARIA widget with roving tabindex is supposed to do:
- Only one item should be tabbable at a time.
- Navigation through items is implemented with arrow keys.
- When tabbing away from the widget and then tabbing into it again, the roving tabindex 'remembers' what the last tabbable item was so that keyboard users land on the last focused item.

However, on first render, _all items are focusable and tabbable_ because none of them has a `tabindex` attribute set, yet.

Ideally, also on first render there should be only one focusable/tabbable item. That is: all the items should have a `tabindex="-1"` attribute except the first one that should have a `tabindex="0"` instead. Optionally, the initial tabbable item should be configurable via a prop.

### Step-by-step reproduction instructions

- Go to the storybook at https://wordpress.github.io/gutenberg/?path=/story/components-experimental-treegrid--default
- The storybook example shows a grid with buttons and input fields.
- Do not interact with any of the items yet.
- Inspect the DOM and observe that none of the buttons / inputs has any `tabindex` attribute.
- Click one of the buttons or one of the inputs.
- Observe the one you clicked gets a `tabindex="0"` attribute.
- Observe all the other buttons and inputs get a `tabindex="-1"` attribute.
- Expected: all the buttons to have a tabindex attribute also on first render.

### Please confirm that you have searched existing issues in the repo.

- [x] Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

- [x] Yes

### Please confirm which theme type you used for testing.

- [ ] Block
- [ ] Classic
- [ ] Hybrid (e.g. classic with theme.json)
- [ ] Not sure

**Comment by Infinite-Null** [comment]
Hi @afercia,

I was successfully able to reproduce the problem in the storybook:

#### Screencast:

https://github.com/user-attachments/assets/8523510d-5343-4931-8ac2-c083019012a9

Would love to work on this and open a PR shortly.
